### DISCUSSION TITLE
网站之后的预计开发路线

### DISCUSSION THREAD

**Discussion by wuyu8512** [LightNovelShelf/Web#86] [Ideas]
### 社区功能：
- [x] 板块的获取，需要分页
- [x] 板块内帖子的获取和发布删除
- [x] 对帖子的回复编辑和删除
- [ ] 每个板块需要有一个独立的实时聊天室

### 编辑器功能：
- [x] 实现HTML和MarkDown编辑器的功能统一，尽可能的实现两者之间代码互相转换
- [x] 实现图片上传
- [ ] 实现附件上传(关键是epub)

### 阅读方面：
- [ ] 发布者可设置关闭评论区
- [x] 实现发布功能，发布者可以设置书籍所需要的阅读等级
- [x] 支持对网站上的书籍一键生成epub，并由发布者选择是否开放给用户下载
- [x] 开发对书籍的统一管理页面，实现书籍基本信息的修改，书籍的删除，章节的新增删除和修改，以及可直接调换章节顺序的功能
- [x] 公告区优化，使用HTMLRead组件
- [x] 时间BUG修复

### 用户方面：
- [ ] 实现登录后的修改密码
- [x] 实现头像修改
- [ ] 登录时如果遇到环境异常，不能直接不让登录，而是应该转为密码+邮箱验证码登录
- [x] 设计一套经验获取方案，使用户可以每日签到，发帖等手段获取经验值用来升级用户等级
- [ ] 可设置隐藏所有书籍的评论区和社区功能
- [x] 消息通知功能，尽可能的实时

### 上面都是对用户可见的一些功能，实际上要搞的东西会更多

**Comment by cyb233** [comment]
可以加个搜索功能的权限等级筛选，类似E站的搜索关键字或下拉菜单应该都可以

**Comment by baka-gourd** [comment]
所有消息建议全部走 SignalR，然后需要一个 API 文档，懒得话可以用 swagger 生成

**Comment by cyb233** [comment]
权限书籍在info页面仍可考虑保持显示权限，与list显示一致或另取位置
